test_gpios.py

```diff
@@ -101,14 +107,28 @@ async def test_interrupts_and_pwm(self, pwm):
 
         starting_count = await interrupt.value()
         await asyncio.sleep(DURATION)
+        should_stop.set()
```

Comment:
im confused by this - if you set it here wouldn't `record_tick_data` return right away?

Reply:
I'd like it to, yeah. but it might be at the top of the for loop, awaiting the next tick. By shutting it down while the ticks are still flowing, it can definitely get into the body of the for loop to notice that it's supposed to shut down. If we turn off the PWM signal and _then_ tell it to shut down, it might hang forever waiting for another tick before it checks whether it ought to return.

I'm reminded of the trouble you had with the numato board, where it wouldn't notice the context was canceled because it was trying to read from the serial port. Same problem here if you stop PWM before shutting down the other coroutine.

Reply:
makes sense thanks for explaining!